azurerm/resource_arm_managed_cluster.go

```diff
@@ -0,0 +1,565 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2017-09-30/containerservice"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedClusterCreate,
+		Read:   resourceArmManagedClusterRead,
+		Update: resourceArmManagedClusterCreate,
+		Delete: resourceArmManagedClusterDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"dns_prefix": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"kubernetes_version": {
+				Type:     schema.TypeString,
+				Required: false,
+				Optional: true,
+			},
+
+			"linux_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admin_username": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"ssh_key": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"key_data": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+				Set: resourceAzureRMManagedClusterLinuxProfilesHash,
+			},
+
+			"agent_pool_profile": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+
+						"count": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      1,
+							ValidateFunc: validateArmManagedClusterAgentPoolProfileCount,
```

Comment:
Could this just use `validation.IntBetween`?

Reply:
Thanks! fixed